### PR TITLE
BDOG-1485 Add team filter to health indicators

### PR DESCRIPTION
**Pull request by hmrc-phil**

### CODE REVIEW

File: app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsController.scala [hunk 1]
```diff
@@ -48,18 +49,26 @@ class HealthIndicatorsController @Inject() (
...
-          r =>
+          formWithErrors => Future.successful(BadRequest(HealthIndicatorsLeaderBoard(Seq.empty, Seq.empty, Seq.empty, formWithErrors))),
+          validForm =>
             for {
-              indicatorsWithTeams <- healthIndicatorsService.findIndicatorsWithTeams(r)
-            } yield Ok(HealthIndicatorsLeaderBoard(indicatorsWithTeams, r, repoName, RepoType.values))
+              indicatorsWithTeams <- healthIndicatorsService.findIndicatorsWithTeams(validForm.repoType.getOrElse(RepoType.Service))
+              teams <- teamsAndRepositoriesConnector.allTeams
+              indicators = indicatorsFilteredByTeam(indicatorsWithTeams, validForm.team)
+            } yield Ok(HealthIndicatorsLeaderBoard(indicators, RepoType.values, teams.sortBy(_.name.asString), form.fill(validForm)))
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsController.scala:63]: `TeamName` already has an implicit ordering. You can just call `.sort`.
Maybe order in `allTeams` function for all clients?

File: app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsService.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ class HealthIndicatorsService @Inject() (
     } yield indicators.map { i =>
       IndicatorsWithTeams(
         i.repoName,
-        owningTeams = repoToTeams.getOrElse(i.repoName, Seq.empty),
+        owningTeams = repoToTeams.getOrElse(i.repoName, Seq.empty).sortBy(_.asString),
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsService.scala:45]: same applies

File: app/views/HealthIndicatorsLeaderBoard.scala.html [hunk 3]
```diff
@@ -58,9 +71,16 @@ <h1>Health Indicator Leaderboard</h1>
                             <a href="/health-indicators/@{indicator.repoName}"><span class="repoName">@indicator.repoName</span></a>
                         </td>
                         <td id="row_@{i}_column_1">
-                            @{if(indicator.owningTeams.size == 1){indicator.owningTeams.map(_.asString)}
-                            else if(indicator.owningTeams.size <=3){indicator.owningTeams.map(_.asString).mkString(", ")}
-                            else("Multiple")}
+                            <ul class="list-group">
+                                @if(indicator.owningTeams.size > 10) {
+                                    <li title="@{indicator.owningTeams.map(_.asString).mkString(", ")}"
+                                    data-toggle="tooltip" data-placement="bottom">Multiple Teams</li>
```
Comment (colin-lamed) [inline app/views/HealthIndicatorsLeaderBoard.scala.html:77]: Should we use the same terminology as `DefaultBranchListPage`? "Various" or "Multiple"?

**Review by colin-lamed — APPROVED** [review]

File: app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsController.scala (hunk 1 above)
Comment (hmrc-phil) [reply app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsController.scala:63]: This is the `Team` object (not `TeamName`) so I can either add implicit ordering by name or cut this down to just use `.sortBy(_.name)`?

File: app/views/HealthIndicatorsLeaderBoard.scala.html (hunk 3 above)
Comment (hmrc-phil) [reply app/views/HealthIndicatorsLeaderBoard.scala.html:77]: This originally just said "Multiple" so happy to revert it back for consistency

**Review by colin-lamed — APPROVED** [review]

**Pull request merged**